### PR TITLE
METRON-835 Use Profiler with Kerberos

### PR DESCRIPTION
**Pull request by nickwallen**
## Contributor Comments

* Enhanced the Kerberos documentation to outline additional steps needed to use the Profiler with Kerberos.  
* Enhanced some of the steps to help others avoid stupid user mistakes that I made when setting up a Kerberized-environment.

## Pull Request Checklist

- [x] Is there a JIRA ticket associated with this PR?
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

### CODE REVIEW

**Comment by mmiklavc** [comment]
You and @JonZeolla are going to collide - https://github.com/apache/incubator-metron/pull/510

**Comment by mmiklavc** [comment]
We would normally want to push format changes to a separate PR because it's hard to follow what has changed here, but the steps still look consistent to me. Were you able to resolve the issues with the Kafka ACLs in the current order? I recall from the mailing list there was some issue there and it wasn't clear if the user needed to be logged in as 'root' instead of 'metron'.

**Comment by nickwallen** [comment]
> We would normally want to push format changes to a separate PR because it's hard to follow what has changed here,

I don't know if we have a normal.  I've seen many instances go both ways during the course of the project.  I can certainly spend the time to split out formatting, if you need me to.  If you don't like the formatting, but want the Kerberos instructions that would be one reason for me to do so.

I had a hard time understanding what we were trying to accomplish with each of the steps, which user the commands would be run under, etc which is why I split them under goal oriented headings.  It also helped me generalize the instructions to run in environments other than Full/Quick Dev.

**Comment by mmiklavc** [comment]
I pulled that from our dev guidelines - https://cwiki.apache.org/confluence/display/METRON/Development+Guidelines

I'm ok with it and I like the new formatting. What was the end result with the Kafka ACL authorization problem you were seeing before? These instructions still have the user creating the ACL's as the 'metron' user - did that work the second time through?

**Comment by nickwallen** [comment]
I think the ACL issue was because there was a loop iterating over the topics setting the ACLs.  The 'yaf' topic was not in that loop so the ACL was never set.  I changed it so none of the validation suggests using 'yaf' since we don't set the ACLs and don't start the parser topology.

**Comment by nickwallen** [comment]
And you are right @mmiklavc .  I should keep formatting changes separate, if not a separate PR, separate commits at least.   Will keep that in mind for the future.

**Comment by mmiklavc** [comment]
+1 by inspection. Thanks for this @nickwallen!

**Comment by nickwallen** [comment]
I have resolved the merge conflicts.  Please take a look @mmiklavc and @JonZeolla

**Comment by nickwallen** [comment]
Conflicts resolved again

**Comment by nickwallen** [comment]
Bump

File: metron-deployment/vagrant/Kerberos-setup.md [hunk 1]
```diff
@@ -55,35 +112,245 @@ General Kerberization notes can be found in the metron-deployment [README.md](..
...
+1. Push some sample data to one of the parser topics. E.g for Bro we took raw data from [incubator-metron/metron-platform/metron-integration-test/src/main/sample/data/bro/raw/BroExampleOutput](../../metron-platform/metron-integration-test/src/main/sample/data/bro/raw/BroExampleOutput)
+
+    ```
+  	cat sample-bro.txt | ${HDP_HOME}/kafka-broker/bin/kafka-console-producer.sh --broker-list ${BROKERLIST}:6667 --security-protocol SASL_PLAINTEXT --topic bro
+  	```
+
+1. Wait a few moments for data to flow through the system and then check for data in the Elasticsearch indices. Replace yaf with whichever parser type you’ve chosen.
+
+    ```
+  	curl -XGET "${ZOOKEEPER}:9200/bro*/_search"
```
Comment (mmiklavc) [inline metron-deployment/vagrant/Kerberos-setup.md]: Just noticed this issue that came in with the original PR - should be ELASTICSEARCH, not ZOOKEEPER.

File: metron-deployment/vagrant/Kerberos-setup.md [hunk 2]
```diff
@@ -55,35 +112,245 @@ General Kerberization notes can be found in the metron-deployment [README.md](..
...
+
+    ```
+  	${METRON_HOME}/bin/start_enrichment_topology.sh
+  	${METRON_HOME}/bin/start_elasticsearch_topology.sh
+  	```
+
+1. Push some sample data to one of the parser topics. E.g for Bro we took raw data from [incubator-metron/metron-platform/metron-integration-test/src/main/sample/data/bro/raw/BroExampleOutput](../../metron-platform/metron-integration-test/src/main/sample/data/bro/raw/BroExampleOutput)
+
+    ```
+  	cat sample-bro.txt | ${HDP_HOME}/kafka-broker/bin/kafka-console-producer.sh --broker-list ${BROKERLIST}:6667 --security-protocol SASL_PLAINTEXT --topic bro
```
Comment (mmiklavc) [inline metron-deployment/vagrant/Kerberos-setup.md]: Env var list above got rid of HDP_HOME. Maybe we can keep both and build KAFKA_HOME from HDP_HOME? Gives the user quick access to other tools also. e.g. zkCli

File: metron-deployment/vagrant/Kerberos-setup.md [hunk 3]
```diff
@@ -1,48 +1,105 @@
...
+1. Deploy a Vagrant development environment; either [Full Dev](full-dev-platform) or [Quick Dev](quick-dev-platform).
+
+1. Export the following environment variables.  These need to be set for the remainder of the instructions. Replace `node1` with the appropriate hosts, if you are running Metron anywhere other than Vagrant.
 
-2. Export env vars. Replace *node1* with the appropriate hosts if running anywhere other than full-dev Vagrant.
     ```
     # execute as root
     sudo su -
     export ZOOKEEPER=node1
     export BROKERLIST=node1
```
Comment (mmiklavc) [inline metron-deployment/vagrant/Kerberos-setup.md]: I've been thinking about this and realized we shouldn't be declaring our zookeeper and kafka broker lists this way because it won't work for anything but a single node environment. The lists are really host:port comma-delimited.

**Comment by nickwallen** [comment]
Thanks @mmiklavc .  I went ahead and addressed all of the items that you mentioned.

**Comment by mmiklavc** [comment]
Oh wow, didn't expect that. Thanks @nickwallen looks great!
